bsoid_app.py

```diff
@@ -6,73 +8,74 @@
-            visuals.show_data_table(raw_input_data, processed_input_data)
-    except NameError:
-        processor = data_preprocess.preprocess()
-        processor.compile_data()
-if st.sidebar.checkbox('Extract and embed features', False, key='f'):
-    [_, _, framerate, _, _, _, processed_input_data, _] = load_data(working_dir, prefix)
-    extractor = extract_features.extract(working_dir, prefix, processed_input_data, framerate)
-    extractor.main()
-if st.sidebar.checkbox('Identify and tweak number of clusters', False, key='c'):
-    [_, sampled_embeddings] = load_embeddings(working_dir, prefix)
-    clusterer = clustering.cluster(working_dir, prefix, sampled_embeddings)
-    clusterer.main()
-if st.sidebar.checkbox('(Optional) What did B-SOiD learn?', False, key='e'):
-    [sampled_features, _] = load_embeddings(working_dir, prefix)
-    [_, assignments, assign_prob, soft_assignments] = load_clusters(working_dir, prefix)
-    exporter = export_training.export(working_dir, prefix, sampled_features,
-                                      assignments, assign_prob, soft_assignments)
-    exporter.save_csv()
-if st.sidebar.checkbox('Create a model', False, key='t'):
-    [features, _] = load_feats(working_dir, prefix)
-    [sampled_features, _] = load_embeddings(working_dir, prefix)
-    [_, assignments, _, _] = load_clusters(working_dir, prefix)
-    learning_protocol = machine_learner.protocol(working_dir, prefix, features, sampled_features, assignments)
-    learning_protocol.main()
-if st.sidebar.checkbox('Generate video snippets for interpretation', False, key='g'):
-    [root_path, data_directories, framerate, pose_chosen, input_filenames, _, processed_input_data, _] \
-        = load_data(working_dir, prefix)
-    [_, _, _, clf, _, _] = load_classifier(working_dir, prefix)
-    creator = video_creator.creator(root_path, data_directories, processed_input_data, pose_chosen,
-                                    working_dir, prefix, framerate, clf, input_filenames)
-    creator.main()
-if st.sidebar.checkbox('Predict old/new files using a model', False, key='p'):
-    [root_path, data_directories, framerate, pose_chosen, input_filenames, _, processed_input_data, _] \
-        = load_data(working_dir, prefix)
-    [_, _, _, clf, _, predictions] = load_classifier(working_dir, prefix)
-    predictor = predict.prediction(root_path, data_directories, input_filenames, processed_input_data, working_dir,
-                                   prefix, framerate, pose_chosen, predictions, clf)
-    predictor.main()
-if st.sidebar.checkbox('Load up analysis app (please close current browser when new browser pops up)', False):
-    streamlit_run('./bsoid_app/bsoid_analysis')
+
+
+processor = data_preprocess.Preprocess(WORKING_DIR,PREFIX,SOFTWARE_CHOICE,FTYPE,ROOT_PATH,FRAMERATE,DATA_DIRECTORIES,POSE_LIST,VALUE)
+processor.compile_data()
+
+[_, _, FRAMERATE, _, _, _, processed_input_data, _] = load_data(WORKING_DIR, PREFIX)
```

Comment:
This is interesting. If `FRAMERATE` is read here, should it be provided by the user at all? 

If it's changed by this call, perhaps `framerate` should be lowercased.